src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -37,6 +37,30 @@ impl ScalarFunc {
         let i = try_opt!(self.children[0].eval_int(ctx, row));
         Ok(Some(Cow::Owned(format!("{:b}", i).into_bytes())))
     }
+
+    pub fn left<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let s = try_opt!(self.children[0].eval_string(ctx, row));
+        let i = try_opt!(self.children[1].eval_int(ctx, row));
```

Comment:
You can cast it here (`as usize`), since you only check it for `usize` based properties.

Reply:
we would check `if i <= 0` the following line